SCons wrongly expects suffix in library include path when building module

**Godot version:** 861d341dffe7c8cd1f6faafffbc51746ac5e3564


**OS/device including version:** *Microsoft Windows 10 Pro* `10.0.17134 Build 17134`




**Issue description:**
When appending a library name to *LIBS* in the SCons environment, e.g. ``env.Append(LIBS=['mysqlcppconn'])`` there results an error when compiling for Windows (haven't tested any other OS): 
>LINK : fatal error LNK1181: cannot open input file 'mysqlcppconn.windows.tools.64.lib'

Instead of finding ``mysqlcppconn.lib``, which exists, it instead looks for the given library with the added suffix ``.windows.tools.64.lib`` in its path.

Renaming ``mysqlcppconn.lib`` in the ``C:\Program Files\MySQL\Connector C++ 1.1\lib\opt`` directory to ``mysqlcppconn.windows.tools.64.lib`` resolves the issue and compiles fine.

**Steps to reproduce:**
Append ``library_name_here``, can be any library, to ``LIBS`` and compile with ``platform=windows``.
Compilation then gives following error:
``[ 97%]  cannot open input file 'library_name_here.windows.tools.lib'``

If you rename ``library_name_here`` to ``library_name_here.windows.tools.lib`` it will compile fine without any related warnings or errors.


**Minimal reproduction project:**
I think it's possible to reproduce with any custom module on Windows where the *SCsub* references an external library. Here's the *SCsub* I used:
```python
Import('env')

env.add_source_files(env.modules_sources,"*.cpp")
env.Append(CPPPATH=['C:/local/boost_1_64_0x64'])
env.Append(LIBPATH=['C:/Program Files/MySQL/Connector C++ 1.1'])
env.Append(LIBS=['mysqlcppconn'])
```
[Zip file containing the custom module](https://github.com/godotengine/godot/files/2173839/mysql.zip)

--- a/SConstruct
+++ b/SConstruct
@@ -409,6 +409,11 @@ if selected_platform in platform_list:
 
     env["PROGSUFFIX"] = suffix + env.module_version_string + env["PROGSUFFIX"]
     env["OBJSUFFIX"] = suffix + env["OBJSUFFIX"]
+    # (SH)LIBSUFFIX will be used for our own built libraries
+    # LIBSUFFIXES contains LIBSUFFIX and SHLIBSUFFIX by default,
+    # so we need to append the default suffixes to keep the ability
+    # to link against thirdparty libraries (.a, .so, .dll, etc.).
+    env["LIBSUFFIXES"] += [env["LIBSUFFIX"], env["SHLIBSUFFIX"]]
     env["LIBSUFFIX"] = suffix + env["LIBSUFFIX"]
     env["SHLIBSUFFIX"] = suffix + env["SHLIBSUFFIX"]
 

